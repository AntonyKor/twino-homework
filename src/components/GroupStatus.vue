<template>
  <div class="group-status">
    <h2>{{ groupName }}</h2>
    <p>{{ Math.round(groupProgress) }}% completed</p>
  </div>
</template>

<script>
export default {
  props: {
    groupProgress: {
      type: Number,
      required: true,
    },
    groupName: {
      type: String,
      default: "",
    },
  },
};
</script>

<style scoped lang="scss">
.group-status {
  margin-top: 30px;
  text-align: center;

  h2 {
    font-size: 1rem;

    @media screen and (min-width: 360px) {
      margin: 0 20px 0 0;
    }
  }

  p {
    margin: 0;
    color: #9e9fa2;
    font-size: 1.3rem;
  }

  @media screen and (min-width: 360px) {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
}
</style>
