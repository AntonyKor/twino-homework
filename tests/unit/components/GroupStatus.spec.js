import { shallowMount } from "@vue/test-utils";
import GroupStatus from "@/components/GroupStatus.vue";

describe("GroupStatus", () => {
  it("should match snapshot without optional props", () => {
    const wrapper = shallowMount(GroupStatus, {
      propsData: { groupProgress: 35 },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should match snapshot with optional props", () => {
    const wrapper = shallowMount(GroupStatus, {
      propsData: { groupProgress: 35, groupName: "Name" },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
