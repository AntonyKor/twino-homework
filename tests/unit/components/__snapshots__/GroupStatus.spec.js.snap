// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupStatus should match snapshot with optional props 1`] = `
<div
  class="group-status"
>
  <h2>
    Name
  </h2>
   
  <p>
    35% completed
  </p>
</div>
`;

exports[`GroupStatus should match snapshot without optional props 1`] = `
<div
  class="group-status"
>
  <h2>
    
  </h2>
   
  <p>
    35% completed
  </p>
</div>
`;
